scanner/src/main/kotlin/scanners/fossid/FossIdScanResults.kt

```diff
@@ -83,7 +84,14 @@ internal fun <T : Summarizable> List<T>.mapSummary(
 
         summary.licences.forEach {
             runCatching {
-                LicenseFinding.createAndMap(it.identifier, location, detectedLicenseMapping = detectedLicenseMapping)
+                val declaredLicense = DeclaredLicenseProcessor.process(setOf(it.identifier))
+                declaredLicense.spdxExpression?.let { spdx ->
```

Comment:
This ignores unmapped licenses. As these could syntax-wise still be valid license IDs (i.e. they do not contain spaces), IMO they should be considered here, too.

Reply:
I don't understand your comment.
When there is an unmapped license, the `spdxExpression` is `null` therefore the `?:` branch will be called: there, the old behavior of calling `LicenseFinding.createAndMap` and potentially raising an `Issue` on failure would still apply.
Am I missing something ?

Reply:
> When there is an unmapped license, the `spdxExpression` is `null`

No, that's not correct, see e.g.

https://github.com/oss-review-toolkit/ort/blob/ab1779038b17773544d6f0ecf436d1f4e9f059d1/utils/ort/src/test/kotlin/DeclaredLicenseProcessorTest.kt#L107-L115

So *both* `spdxExpression` *and* `unmapped` may contain (disjunctive) values.

Not imagine the test would not use the string "invalid" but "ALicenseNameOnlyUsedbyFossId". The current implementation would swallow "ALicenseNameOnlyUsedbyFossId", but my guess is that you'd want to keep it, although it's neither an SPDX (core) license ID nor a `LicenseRef-`.

Reply:
First, please not that the `DeclaredLicenseProcessor.process` is called with only one license in both cases. 
Therefore, if `ALicenseNameOnlyUsedbyFossId` is given to `process`. It is EITHER in `spdxExpression` or in `unmapped`.
I cannot land in both.

If this is in `unmapped`, the `spdxExpression` will be `null`. Hence the `?:` branch will be called, this will then call `spdxExpression`. The latter does a `toSpdx`.
Therefore finally, the `onFailure` will be called which will create an issue.

I don't see what else can be done ?!

Reply:
> I don't see what else can be done ?!

My proposal is to make the `DeclaredLicenseProcessor.process()` overload that only takes a single string public, and then simply use

    val license = snippet.artifactLicense?.let { DeclaredLicenseProcessor.process(it) }
        ?: SpdxConstants.NOASSERTION.toSpdx()

This code could also be moved to before the line `val ortSnippet = OrtSnippet(` to be closer to its use.

That addressed several issues with the proposed code:
- It removes complexity only needed for compound expressions, which cannot occur here.
- `DeclaredLicenseProcessor.process()` doe snot throw, so the `onFailure` handling is unneeded
- For the use in line 135, it avoids creating a `LicenseFinding` from which only `license` is being used.